### ISSUE TITLE
Why fairseq-interactive and fairseq-genetate generate results so different? 

### ISSUE THREAD

**Issue by kangcp** [pytorch/fairseq#2712]
## ❓ Questions and Help
Why fairseq-interactive and fairseq-genetate generate results so different? 

### Before asking:   
1. search the issues.   
2. search the docs.    

#### What is your question?
I load the same model and use these two methods to produce completely different results. Is it my setting wrong?
#### Code

* train:
GPU_ID=0,1,2,3
DATA_BIN_DIR="asr_dataset/bin/"
OUT_DIR="asr_dataset/models/"
BATCH_SIZE=64
MAX_LEN=400
MAX_TOKENS=4096
SEED=1
# MAX_UPDATE=300000 #default
# MAX_UPDATE=10
MAX_EPOCH=100

CUDA_VISIBLE_DEVICES="${GPU_ID}" fairseq-train \
    ${DATA_BIN_DIR} \
    --save-dir ${OUT_DIR} \
    --ddp-backend=no_c10d \
    --task translation_lev \
    --criterion nat_loss \
    --arch levenshtein_transformer \
    --noise random_delete \
    --share-all-embeddings \
    --optimizer adam --adam-betas '(0.9,0.98)' \
    --lr 0.0005 --lr-scheduler inverse_sqrt \
    --min-lr '1e-09' --warmup-updates 10000 \
    --warmup-init-lr '1e-07' --label-smoothing 0.1 \
    --dropout 0.3 --weight-decay 0.01 \
    --decoder-learned-pos \
    --encoder-learned-pos \
    --apply-bert-init \
    --log-format 'simple' --log-interval 100 \
    --fixed-validation-seed ${SEED} \
    --max-tokens ${MAX_TOKENS} \
    --save-interval-updates 10000 \
    --max-epoch ${MAX_EPOCH} \
    --fp16

* fairseq-genetate
DATA_BIN_DIR="asr_dataset/bin/"
MODEL="asr_dataset/models/checkpoint_best.pt"
BEAM=2
BATCH_SIZE=64

fairseq-generate \
    ${DATA_BIN_DIR} \
    --gen-subset valid \
    --task translation_lev \
    --path ${MODEL} \
    --iter-decode-max-iter 9 \
    --iter-decode-eos-penalty 0 \
    --beam ${BEAM} --remove-bpe \
    --print-step \
    --batch-size ${BATCH_SIZE}

* fairseq-interactive
DATA_BIN_DIR="./asr_dataset/bin/"
MODEL="./asr_dataset/models/checkpoint_best.pt"
BEAM=2
BATCH_SIZE=64
input=$1

CUDA_VISIBLE_DEVICES=0 fairseq-interactive \
    ${DATA_BIN_DIR} \
    --task translation_lev \
    --path ${MODEL} \
    --iter-decode-max-iter 9 \
    --iter-decode-eos-penalty 0 \
    --beam ${BEAM} --remove-bpe \
    --print-step \
    --buffer-size ${BATCH_SIZE} \
    --input ${input} \
    --fp16

$1: 
cat input.txt:
 想 吃 海 鲜
 我 很 快 乐

   
output by fairseq-generate:
1.  想 吃 海 鲜
2. 我 很 快 乐

output by fairseq-interactive :
1. 我 的
2. 我

This result is very strange

#### What have you tried?
My issue is different with [#568](https://github.com/pytorch/fairseq/issues/568)
 
I also tried:

fr2en = TransformerModel.from_pretrained(
        model_name_or_path=model_path,
        checkpoint_file='checkpoint_best.pt',
        data_name_or_path=user_dir,
        task='translation_lev',
        iter_decode_max_iter=9,
        iter_decode_eos_penalty=0,
        beam=2,
        buffer_size=64,
        fp16=True,
        print_step=True
)
fr2en.eval()
fr2en.translate('想 吃 海 鲜')

but the result is same with  fairseq-interactive

#### What's your environment?

 - fairseq Version (e.g., 1.0 or master): 0.9.0
 - PyTorch Version (e.g., 1.0): 1.3.1
 - OS (e.g., Linux): Ubuntu
 - How you installed fairseq (`pip`, source): pip
 - Build command you used (if compiling from source):
 - Python version:  python3.6.9
 - CUDA/cuDNN version: 10.0
 - GPU models and configuration: GeForce RTX 2080 Ti
 - Any other relevant information:
